plugins/link/dialogs/link.js

```diff
@@ -180,6 +181,22 @@
                                         );
                                         a.email.address=this.getValue()
                                     }
+                                    , validate: function() {
+                                        var a = this.getDialog();
+                                        if (a.getValueOf("info", "linkType") == 'email') {
+                                            const EMAIL_REGEXP = /(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/;
```

Comment:
Where did we get this regex from? After testing it, it does match e-mail addresses, but I'm curious about the special characters and references like \x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f

Reply:
From pxIsEmail Filter from Presto to keep the check consistent.